### ISSUE TITLE
Calling `.toPandas()` on a DataFrame

### ISSUE THREAD

**Issue by dserban** [Morphl-AI/MorphL-Model-Publishers-Churning-Users#21]
The `.toPandas()` method call here:

https://github.com/Morphl-AI/MorphL-Model-Publishers-Churning-Users/blob/bug-fix/remove-dask-and-fix-window-function/training/model_generator/ga_chp_model_generator.py#L42-L51

Calling `.toPandas()` on a DataFrame negates the distributed architecture and the parallel execution that Spark provides by forcing all the data in a DataFrame to be transferred from the executors to the driver. It is essentially a DDOS attack launched by the Spark worker machines sending all the data under their control, to the Spark master machine, all at the same time.

The DataFrame methods `.collect()` and `.toPandas()` are designed to only be used after you distilled a large dataframe (10 billion rows for example) to a very small, bounded result (let's say you compute some top 10 list). It doesn't make sense to use them on something that is either a training set, a test set or a validation set.

**Comment by dserban** [comment]
Oh, OK. I thought the process of transitioning the pipeline to be 100% `pyspark` would also fix this as a side effect.
In any case, it's good to keep in mind for future reference that without fixing this, the entire pipeline cannot be considered a big data product.
